### PR TITLE
Update description component to match refined mocks. Refactor. #84.

### PR DESCRIPTION
**Pull request by frano-m**
### Reviewers

#84 

@MillenniumFalconMechanic 

### Changes
- modified description config to single component
- added `Description` story
- added `projectMapper` utility
- refactored `Section` and all related section components to project folder
- refactored `Section` prop `title` from optional to required

### CODE REVIEW

File: explorer/app/components/Project/components/common/projectMapper.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// App dependencies
```
Comment (MillenniumFalconMechanic) [inline explorer/app/components/Project/components/common/projectMapper.ts:1]: Thank you for going ahead and splitting out the binding of the API response from the building of the view model!

Can we keep the naming convention of the binding of the API response as `xTransformer` (so in this case, `projectTransformer`) and then rename the current `projectTransformer` ([here](https://github.com/clevercanary/data-browser/blob/84-detail-description-refactor/explorer/site-config/hca-dcp/dev/projectTransformer.ts)) to be `projectViewModelBuilder`?

File: explorer/app/components/Project/components/common/projectMapper.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import { ProjectResponse } from "../../../../models/responses";
+
+/**
+ * Maps project description.
+ * @param project
+ */
+export function mapProjectDescription(
```
Comment (MillenniumFalconMechanic) [inline explorer/app/components/Project/components/common/projectMapper.ts:8]: What do you think about keeping the naming convention simple here and call this `getProjectDescription`?

File: explorer/app/components/Project/components/common/projectMapper.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// App dependencies
+import { ProjectResponse } from "../../../../models/responses";
+
+/**
+ * Maps project description.
```
Comment (MillenniumFalconMechanic) [inline explorer/app/components/Project/components/common/projectMapper.ts:5]:
```suggestion
 * Maps project description from API response.
```

File: explorer/app/components/Project/components/common/projectMapper.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+ * @param project
```
Comment (MillenniumFalconMechanic) [inline explorer/app/components/Project/components/common/projectMapper.ts:6]:
```suggestion
 * @param project - Project response model return from API.
```

**Review by MillenniumFalconMechanic — CHANGES_REQUESTED** [review]
Very clean, thanks F! I have asked for a couple of name changes; let me know your thoughts.

File: explorer/app/components/Project/components/common/projectMapper.ts (hunk 1 above)
Comment (frano-m) [reply explorer/app/components/Project/components/common/projectMapper.ts:1]: Great name - thank you!

**Review by MillenniumFalconMechanic — APPROVED** [review]
LGTM, thanks F!

**Pull request merged**